lib/extendedkey.ex

```diff
@@ -0,0 +1,621 @@
+defmodule Bitcoinex.ExtendedKey do
+  @moduledoc """
+  	Contains an an extended key as documented in BIP 32.
+  """
+  alias Bitcoinex.Secp256k1.{Params, Point, PrivateKey, Math}
+  alias Bitcoinex.Base58
+
+  use Bitwise, only_operators: true
+
+  defmodule DerivationPath do
+    @moduledoc """
+    Contains a list of integers (or the :any atom) representing a bip32 derivation path. 
+    The :any atom represents a wildcard in the derivation path. DerivationPath structs can
+    be used by ExtendedKey.derive_extended_key to derive a child key at the given path.
+    """
+
+    @min_non_hardened_child_num 0
+    # 2^31 - 1
+    @max_non_hardened_child_num 0x80000000 - 1
+    @min_hardened_child_num 0x80000000
```

Comment:
We could simplify these constants to just include `min_hardened_child_num` and `max_hardened_child_num`, the other ones do not seem to be useful

Reply:
Kafai and I discussed this on his PR. he thought it would provide more clarity